src/app/page.tsx

```diff
@@ -2,22 +2,20 @@ import { Events } from '../components/Events';
 import { getAllEvents } from '../api/events';
 import { Pagination } from '../components/Pagination';
 import { PAGINATION } from '../utils/constants';
+import { getFirstQueryParam, parseToNumber } from '../utils/functions';
 
 const Home = async ({
   searchParams,
 }: {
   searchParams: { [key: string]: string | string[] | undefined };
 }) => {
-  const page: string =
-    typeof searchParams?.page === 'string'
-      ? searchParams.page
-      : PAGINATION.PAGE;
-  const limit: string =
-    typeof searchParams?.limit === 'string'
-      ? searchParams.limit
-      : PAGINATION.LIMIT;
+  const page = getFirstQueryParam(searchParams?.page, PAGINATION.PAGE);
+  const limit = getFirstQueryParam(searchParams?.limit, PAGINATION.LIMIT);
```

Comment:
I left it like this... If you have better solution please be concrete. 😅 I do not know what you want.